modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/PaginationBarTag.java

```diff
@@ -0,0 +1,318 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.frontend.taglib.clay.servlet.taglib;
+
+import com.liferay.frontend.taglib.clay.internal.servlet.taglib.BaseContainerTag;
+import com.liferay.frontend.taglib.clay.servlet.taglib.util.PaginationBarDelta;
+import com.liferay.frontend.taglib.clay.servlet.taglib.util.PaginationBarLabels;
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.language.LanguageUtil;
+import com.liferay.portal.kernel.util.StringUtil;
+import com.liferay.taglib.util.TagResourceBundleUtil;
+
+import java.util.List;
+import java.util.Map;
+import java.util.ResourceBundle;
+
+import javax.servlet.jsp.JspException;
+import javax.servlet.jsp.JspWriter;
+
+/**
+ * @author Julien Castelain
+ */
+public class PaginationBarTag extends BaseContainerTag {
+
+	@Override
+	public int doStartTag() throws JspException {
+		setAttributeNamespace(_ATTRIBUTE_NAMESPACE);
+
+		PaginationBarLabels labels = getLabels();
+
+		if (labels == null) {
+			setLabels(
+				new PaginationBarLabels(
+					"Showing {0} to {1} of {2}", "{0} items", "{0} items"));
+		}
+
+		return super.doStartTag();
+	}
+
+	public Integer getActiveDelta() {
+		return _activeDelta;
+	}
+
+	public Integer getActivePage() {
+		return _activePage;
+	}
+
+	public List<PaginationBarDelta> getDeltas() {
+		return _deltas;
+	}
+
+	public List<Integer> getDisabledPages() {
+		return _disabledPages;
+	}
+
+	public Integer getEllipsisBuffer() {
+		return _ellipsisBuffer;
+	}
+
+	public PaginationBarLabels getLabels() {
+		return _labels;
+	}
+
+	public Integer getTotalItems() {
+		return _totalItems;
+	}
+
+	public void setActiveDelta(Integer activeDelta) {
+		_activeDelta = activeDelta;
+	}
+
+	public void setActivePage(Integer activePage) {
+		_activePage = activePage;
+	}
+
+	public void setDeltas(List<PaginationBarDelta> deltas) {
+		_deltas = deltas;
+	}
+
+	public void setDisabledPages(List<Integer> disabledPages) {
+		System.out.println("Disabled pages set");
```

Comment:
We should not use System.out.println :)

Reply:
Yeah, sorry that was me debugging

Reply:
Updated in [`5e1dc7b`](https://github.com/liferay-frontend/liferay-portal/pull/1224/commits/5e1dc7b5d700ebff7ab18087ece50cd2619e41d6)